File: 2/2.py
def solve_a(data, rules_win, rules_draw, score_form):
    scores = []
    for they, I in data:
        score = 0
        if I == rules_win[they]:
            score += 6
        elif I == rules_draw[they]:
            score += 3

        score += score_form[I]
        scores.append(score)
    return sum(scores)

def solve_b(data, rules_win, rules_draw, rules_lose, score_form):
    scores = []
    for they, I in data:
        score = 0
        if I == 'X':
            my_choice = rules_lose[they]
            score += score_form[my_choice]
        elif I == 'Y':
            my_choice = rules_draw[they]
            score += 3
            score += score_form[my_choice]
        else:
            my_choice = rules_win[they]
            score += 6
            score += score_form[my_choice]
        scores.append(score)
    return sum(scores)


d_win = {
    'C': 'X',
    'A': 'Y',
    'B': 'Z'
}

d_draw = {
    'A': 'X',
    'B': 'Y',
    'C': 'Z'
}

d_lose = {
    'A': 'Z',
    'B': 'X',
    'C': 'Y'
}
scores_per_choice = {
    'X': 1,
    'Y': 2,
    'Z': 3
}



if __name__ == "__main__":
    data = [line.strip('\n').split(' ') for line in open('input.txt', 'r').readlines()]
    
    a = solve_a(data, d_win, d_draw, scores_per_choice)
    b = solve_b(data, d_win, d_draw, d_lose, scores_per_choice)
    print(a, b)

    
        


        

File: 2/2.jl
# A
let 
f = readlines(open("input.txt"))

wins = ["A Y", "B Z", "C X"]
draws = ["A X", "B Y", "C Z"]


scores = Dict{Char, Integer}('X' => 1, 'Y' => 2, 'Z' => 3)
total_score = 0

for line in f
    opponent, me = line[1], line[3]
    total_score += scores[me]

    if line in wins
        total_score += 6
    elseif line in draws
        total_score += 3
    end

end
a = total_score
println(a)
end

# b
let 
f = readlines(open("input.txt"))

wins = ["A Y", "B Z", "C X"]
draws = ["A X", "B Y", "C Z"]
loses = ["A Z", "B X", "C Y"]

scores = Dict{Char, Integer}('X' => 1, 'Y' => 2, 'Z' => 3)
total_score = 0

for line in f
    opponent, result = line[1], line[3]
    if result == 'X'
        for choice in loses
            if choice[1] == opponent
                me = choice[3]
                total_score += scores[me]
            end
        end
    elseif result == 'Y'
        total_score += 3
        for choice in draws
            if choice[1] == opponent
                me = choice[3]
                total_score += scores[me]
            end
        end
    else
        total_score += 6
        for choice in wins
            if choice[1] == opponent
                me = choice[3]
                total_score += scores[me]
            end
        end
    end   
end

b = total_score
println(b)
end
